pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java

```diff
@@ -194,6 +197,38 @@ protected static void deleteRecursive(BaseResources resources, final String path
         }
     }
 
+    protected static CompletableFuture<Void> deleteRecursiveAsync(BaseResources resources, final String pathRoot) {
+        PathUtils.validatePath(pathRoot);
+        List<String> tree = null;
+        try {
+            tree = listSubTreeBFS(resources, pathRoot);
+        } catch (MetadataStoreException e) {
+
```

Comment:
We could ignore this exception, right? Could you add some explanation?

Reply:
Yes, we can ignore this exception.